tests/test_link_io_structure.py

```diff
@@ -0,0 +1,121 @@
+from symlink_komodo.sanity_check import read_link_structure, compare_links, check_root_nodes, verify_integrity
+from symlink_komodo.create_links import create_symlinks
+
+import os
+from shutil import copy
+import json
+
+
+def test_read_folder_structure(tmpdir):
+    with tmpdir.as_cwd() as old_dir:
+        os.mkdir('2012.01.rc1')
+        os.mkdir('2012.01.rc2')
+        os.mkdir('bleeding')
+        os.mkdir('2012.01.12')
+
+        os.symlink('2012.01.12', '2012.01')
+        os.symlink('2012.01', 'stable')
+        os.symlink('2012.01', 'testing')
+        os.symlink('2012.01.rc2', 'unstable')
+
+        expected_result = {
+            'root_folder': tmpdir,
+            'root_links' : [
+                'stable',
+                'unstable',
+                'testing'
+            ],
+            'links': {
+                '2012.01': '2012.01.12',
+                'stable': '2012.01',
+                'testing': '2012.01',
+                'unstable': '2012.01.rc2'
+            }
+        }
+
+        output_dict = read_link_structure(os.getcwd())
+
+        assert type(output_dict) == dict
+        assert compare_links(output_dict, expected_result)
+
+
+def test_create_symlinks(tmpdir):
+    links_dict = {
+        'root_folder': tmpdir,
+        'links': {
+            '2012.01': '2012.01.12',
+            'stable': '2012.01',
+            'testing': '2012.01',
+            'unstable': '2012.01.rc2'
+        }
+    }
+
+    with tmpdir.as_cwd() as old_dir:
+        os.mkdir('2012.01.12')
+        os.mkdir('2012.01.rc2')
+        create_symlinks(links_dict)
+
+        assert os.path.exists('stable')
+        assert os.path.islink('stable')
+        assert os.readlink('stable') == '2012.01'
+        assert os.path.realpath('stable') == os.path.realpath('2012.01.12')
+        assert os.path.realpath('unstable') == os.path.realpath('2012.01.rc2')
+
+
+def test_integration(tmpdir):
+    
+    test_file = os.getenv('PYTEST_CURRENT_TEST')
+    test_folder = os.path.dirname(test_file)
+    copy(os.path.join(test_folder, 'data/links.yml'), tmpdir)
+    with tmpdir.as_cwd() as old_dir:
+        input_dict = json.load(open('links.yml'))
+        
+        input_dict['root_folder'] = os.path.realpath(input_dict['root_folder'])
+
+        os.mkdir(input_dict['root_folder'])
+        os.mkdir(os.path.join(input_dict['root_folder'], '2012.01.12'))
+        os.mkdir(os.path.join(input_dict['root_folder'], '2012.01.rc1'))
+
+        create_symlinks(input_dict)
+
+        output = read_link_structure('.')
+        assert compare_links(input_dict, output)
+
+
+def test_root_links(tmpdir):
+    
+    test_file = os.getenv('PYTEST_CURRENT_TEST')
+    test_folder = os.path.dirname(test_file)
+    copy(os.path.join(test_folder, 'data/links.yml'), tmpdir)
+    with tmpdir.as_cwd() as old_dir:
+        os.chdir(tmpdir)
+        input_dict = json.load(open('links.yml'))
+        
+        check_root_nodes(input_dict)
```

Comment:
How about naming this one `assert_root_nodes` and then raising an `AssertionError` instead?

Reply:
:+1: